js/DataService.js

```diff
@@ -8,13 +8,46 @@ angular.module('PathOfDamage')
     var int = parseInt(string);
     return isNaN(int) ? null : int;
   }
+  
+  function Row(key, value) {
+    this.enabled = true;
+    this.name = '';
+    this.value = null;
+    if (key !== undefined) {
```

Comment:
I guess coming from java it seems like that's why you subclass, but in javascript subclassing is a bit messier and subclassing to add one additional variable doesn't end up looking very clean. It seems odd to be passing in one optional key/value pair and a bit hard to follow? I could be wrong.